src/world/partner/watt.c

```diff
@@ -1,29 +1,169 @@
 #include "common.h"
+#include "effects.h"
+#include "../src/world/partners.h"
 
-INCLUDE_ASM(s32, "world/partner/watt", func_802BD100_31CC70);
+typedef struct unk_802BE310_C {
+    /* 0x00 */ char unk_00[0x28];
+    /* 0x28 */ s32 unk_28;
+} unk_802BE310_C;
 
-INCLUDE_ASM(s32, "world/partner/watt", func_802BD180_31CCF0);
+typedef struct unk_802BE310 {
+    /* 0x00 */ char unk_00[0x0C];
+    /* 0x0C */ unk_802BE310_C* unk_0C;
+    /* 0x10 */
+} unk_802BE310;
+
+typedef struct unk_802BE274_31DDE4 {
+    /* 0x00 */ char unk_00[4];
+    /* 0x04 */ s32 unk_04;
+    /* 0x08 */ s32 unk_08;
+} unk_802BE274_31DDE4;
+
+extern s32 D_802BE300;
+extern s32 D_802BE304;
+extern s32 D_802BE308;
+extern s32 D_802BE30C;
+extern unk_802BE310* D_802BE310;
+extern unk_802BE274_31DDE4* D_802BE274_31DDE4;
+
+void func_802BD100_31CC70(Npc* npc) {
+    if (!(npc->flags & 2)) {
+        if (D_802BE310 == NULL) {
+            D_802BE310 = (unk_802BE310*)playFX_73(0, npc->pos.x, npc->pos.y + 13.0f, npc->pos.z, 0.9f, 5, 0);
+        }
+    }
+}
+
+s32 func_802BD180_31CCF0(void) {
+    if (D_802BE310 != NULL) {
+        D_802BE310->unk_0C->unk_28 = 1;
+        D_802BE310 = NULL;
+        return 1;
+    }
+}
 
 INCLUDE_ASM(s32, "world/partner/watt", func_802BD1AC_31CD1C);
 
-INCLUDE_ASM(s32, "world/partner/watt", func_802BD23C_31CDAC);
+void func_802BD23C_31CDAC(Npc* npc) {
+    npc->collisionHeight = 24;
+    npc->collisionRadius = 24;
+    D_802BE308 = 0;
+    D_802BE300 = 0;
+    D_802BE30C = 0;
+    D_802BE304 = 20;
+    D_802BE310 = NULL;
+}
+
+s32 func_802BD27C_31CDEC(Evt* script, s32 isInitialCall) {
+    Npc* npc = script->owner2.npc;
+
+    if (isInitialCall != 0) {
+        partner_init_get_out(npc);
+    }
+    return partner_get_out(npc) != FALSE;
```

Comment:
You should be able to turn this into an if/else that returns ApiStatus_BLOCK and ApiStatus_DONE1.